frontend/spec/features/first_order_promotion_spec.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.feature "First Order promotion" do
+  given!(:promotion) do
+    FactoryBot.create(
+      :promotion_with_first_order_rule,
+      :with_order_adjustment,
+      code: "FIRSTONEFREE",
+      per_code_usage_limit: 10
+    )
+  end
+
+  background do
+    create(:store)
+    product = FactoryBot.create(:product)
+    visit spree.root_path
+    click_link product.name
+    click_button "Add To Cart"
+  end
+
+  scenario "Adding first order promotion to cart and checking out as guest" do
+    fill_in "Coupon code", with: "FIRSTONEFREE"
+    click_button "Update"
```

Comment:
Now that #2958 is merged, you can use `click_button "Apply Code"` to take advantage of the new endpoints and avoid deprecation warnings :tada:

Reply:
Updated